sashimi-webapp/src/logic/filemanager/file.js

```diff
@@ -1,21 +1,53 @@
-import storage from '../../database/storage';
+import storage from 'src/database/storage';
 import Folder from './folder';
 
+/* Constants */
+
+// const RENAME_ERROR_MSG = `Another file in "${this.parentFolder.path}" has the same file name`;
+const RENAME_ERROR_MSG = 'Another file in the current folder has the same name';
+const MOVE_SAME_FOLDER_ERROR_MSG = 'Attempting to move to current folder';
+const MOVE_INVALID_FODLER_ERROR_MSG = 'Attempting to move to an invalid folder';
+
+/* Private Functions */
+
+function isCurrentFolder(destFolder) {
+  return destFolder.id === this.parentFolder.id;
+}
+
+function isInvalidFolder(destFolder) {
+  if (Folder.getFolder(destFolder.id) === undefined) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+function hasSameFileName(newFileName) {
+  const currParentFolder = this.parentFolder;
```

Comment:
Extra care is needed when calling a "non-instance function" that make uses of `this` inside an `instance function`.
It is necessary to check what the `this` in this function is actually referring to, see [javascript-type-of-custom-object](http://stackoverflow.com/questions/1486300/javascript-type-of-custom-object).

Unfortunately, in your use case here, the `this` is actually `undefined` when it is called from `Folder.prototype.rename`.
This is actually related to another issue: [why-is-this-undefined-inside-class-method-when-using-promises](http://stackoverflow.com/questions/34930771/why-is-this-undefined-inside-class-method-when-using-promises)

See the comment I leave at `Folder.prototype.rename` to see how to fix this problem.